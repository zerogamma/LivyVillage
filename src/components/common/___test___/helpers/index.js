import React from 'react'
import renderer from 'react-test-renderer'
import theme from '../../../../utils/style/theme'
import { ThemeProvider } from 'styled-components'

export function renderWithTheme(component) {
    return renderer.create(
        <ThemeProvider theme={theme}>{component}</ThemeProvider>
    )
}
