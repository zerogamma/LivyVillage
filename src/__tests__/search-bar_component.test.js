import React from 'react'
import SearchBar from '../components/village/search-bar'

import 'jest-styled-components'
import { renderWithTheme } from '../components/common/___test___/helpers'

describe('SearchBar Component render test', () => {
    it('renders SearchBar component correctly', () => {
        const data = 'testing'
        const comp = renderWithTheme(
            <SearchBar search={data} placeholder="testing-component" />
        ).toJSON()
        expect(comp).toMatchSnapshot()
    })
})
