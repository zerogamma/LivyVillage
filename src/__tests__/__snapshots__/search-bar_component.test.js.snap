// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar Component render test renders SearchBar component correctly 1`] = `
.c0 {
  text-align: left;
  margin-left: 35px;
}

.c1 {
  width: 250px;
  border-radius: 5px;
  text-align: center;
}

.c2 {
  cursor: pointer;
  position: absolute;
  margin-left: 5px;
  margin-top: 5px;
}

.c3 {
  cursor: pointer;
  position: absolute;
  margin-left: 30px;
  font-size: 2rem !important;
}

@media (max-width:767px) {
  .c0 {
    margin-left: 18px;
  }
}

@media (max-width:767px) {
  .c1 {
    width: 190px;
  }

  .c1 .MuiInputBase-input {
    font-size: 12px;
  }
}

<div
  className="c0"
>
  <div
    className="MuiFormControl-root MuiTextField-root c1"
    onKeyPress={[Function]}
  >
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="testing-component"
        required={false}
        type="text"
      />
    </div>
  </div>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root c2"
    focusable="false"
    onClick={[Function]}
    viewBox="0 0 24 24"
  >
    <path
      d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
    />
  </svg>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root c3"
    focusable="false"
    onClick={[Function]}
    viewBox="0 0 24 24"
  >
    <path
      d="M5 13h14v-2H5v2zm-2 4h14v-2H3v2zM7 7v2h14V7H7z"
    />
  </svg>
</div>
`;
